www/class/centreonAuth.class.php

```diff
@@ -213,6 +213,16 @@ protected function checkPassword($password, $token = "", $autoImport = false)
                 } elseif ($this->passwdOk == 1) {
                     if (isset($this->ldap_store_password[$arId]) && $this->ldap_store_password[$arId]) {
                         if (!isset($this->userInfos["contact_passwd"])) {
+                            // Retrieving the created contact_id
+                            $res = $this->pearDB->prepare(
+                                "SELECT contact_id FROM contact
+                                WHERE ar_id = :arId"
+                            );
+                            $res->bindValue(':arId', $arId, \PDO::PARAM_STR);
+                            $res->execute();
+                            $row = $res->fetch();
+                            $this->userInfos['contact_id'] = $row['contact_id'];
```

Comment:
you should create a method `updatePasswordByAlias` in centreonContact.class.php
arguments will be `$this->login` and `$hashedPassword`

furthermore, your current query can return multiple contacts if they are coming from the same ldap server

Reply:
function added